### DISCUSSION TITLE
How to get a drive by name - filter is being ignored?

### DISCUSSION THREAD

**Discussion by emerbrito** [microsoftgraph/msgraph-sdk-dotnet#2643] [General]
I have a site ID, now I'm trying to return a drive named "Account".
No matter how I try it I get all the drives under the site, the filter is never applied.

Here is an example:

```csharp
var result = await graphClient
    .Sites[sharePointSiteId]
    .Drives
    .GetAsync(c =>
    {
        c.QueryParameters.Filter = "name eq 'Account'";
    });
```
All drives are returned:

<img width="510" alt="image" src="https://github.com/user-attachments/assets/a9ca045a-e4ea-4898-b97d-4dc7d06cd590">

Different approach, same outcome:

```csharp
var request = graphClient
    .Sites[sharePointSiteId]
    .Drives
    .ToGetRequestInformation(c => c.QueryParameters.Filter = "name eq 'Opportunity'");

var results = await graphClient.RequestAdapter.SendAsync(request, DriveCollectionResponse.CreateFromDiscriminatorValue);
```
Is this by design, an issue or am I doing something wrong?
Tested with 5.56.1 (latest stable to date) and 5.78.1 (preview).

**Comment by andrueastman** [comment]
Thanks for raising this @emerbrito 

Last I checked `$filter` is supported in limited scenarios in drive. The docs also do not list `$filter` as a supported query for the path. 
https://learn.microsoft.com/en-us/graph/api/drive-list?view=graph-rest-1.0&tabs=http#optional-query-parameters

I would suggest asking a question at https://aka.ms/askGraph on more information on this support. And possibly file a feature request at 
https://aka.ms/graphFeedback
